packages/shared/src/hooks/useRegistration.ts

```diff
@@ -54,11 +55,25 @@ const useRegistration = ({
   const { displayToast } = useToastNotification();
   const { trackingId, referral, referralOrigin } = useContext(AuthContext);
   const timezone = getUserDefaultTimezone();
-  const { data: registration, isLoading: isQueryLoading } = useQuery(
-    key,
-    () => initializeKratosFlow(AuthFlow.Registration),
-    { refetchOnWindowFocus: false },
-  );
+  const {
+    data: registration,
+    isLoading: isQueryLoading,
+    error,
+  } = useQuery(key, () => initializeKratosFlow(AuthFlow.Registration), {
+    refetchOnWindowFocus: false,
+  });
+
+  useEffect(() => {
+    if (error) {
+      trackEvent({
+        event_name: AuthEventNames.RegistrationInitialisationError,
+        extra: JSON.stringify({
+          error,
+          origin: Origin.InitializeRegistrationFlow,
+        }),
+      });
+    }
+  }, [error, trackEvent]);
```

Comment:
We should remove `trackEvent` here right, since it is not stable and as we do until we sort out tracking hook. Disable lint is ok here but if you wanna avoid even that we can move it to `queryFn` and catch there, something like this:

```jsx
const { data: registration, isLoading: isQueryLoading } = useQuery(
  key,
  () => {
    try {
      const result = await initializeKratosFlow(AuthFlow.Registration);

      return result;
    } catch (error) {
      trackEvent({
        event_name: AuthEventNames.RegistrationInitialisationError,
        extra: JSON.stringify({
          error,
          origin: Origin.InitializeRegistrationFlow,
        }),
      });

      throw error;
    }
  },
  { refetchOnWindowFocus: false },
);
```

Reply:
Cool let me disable it for now until we sort the hook